tests/utils/test_train_utils.py

```diff
@@ -165,6 +166,27 @@ def test_warn_deprecated_model_confidences():
         train_utils._check_confidence_setting(component_config)
 
 
+@pytest.mark.parametrize(
+    "component_config, raises_exception",
+    [
+        ({TOLERANCE: 0.5}, False),
+        ({TOLERANCE: 0.0}, False),
+        ({TOLERANCE: 1.0}, False),
+        ({TOLERANCE: -1.0}, True),
+        ({TOLERANCE: 2.0}, True),
+        ({}, False),
+    ],
+)
+def test_tolerance_setting(
```

Comment:
nit: Should be really to separate test functions (https://dzone.com/articles/unit-testing-best-practices-how-to-get-the-most-ou, Unit Tests Should Verify a Single-Use Case)

Reply:
Do you mean that the empty `{}` case should not be added to the test cases here?

Reply:
yeah I wouldn't do so (it's a `nit` so fine for now, it's more on a general note)